heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -97,60 +101,152 @@ const JEST_CONFIG_PACKAGE_FOLDER: string = path.dirname(require.resolve('jest-co
 /**
  * @internal
  */
-export class JestPlugin implements IHeftPlugin<IJestPluginOptions> {
-  public readonly pluginName: string = PLUGIN_NAME;
+export default class JestPlugin implements IHeftTaskPlugin<IJestPluginOptions> {
+  private static _jestConfigurationFileLoader: ConfigurationFile<IHeftJestConfiguration> | undefined;
+
   public readonly optionsSchema: JsonSchema = JsonSchema.fromFile(PLUGIN_SCHEMA_PATH);
 
   /**
-   * Runs required setup before running Jest through the JestPlugin.
+   * Setup the hooks and custom CLI options for the Jest plugin.
+   *
+   * @override
+   */
+  public apply(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration,
+    pluginOptions?: IJestPluginOptions
+  ): void {
+    // Flags
+    const detectOpenHandles: CommandLineFlagParameter = taskSession.parametersByLongName.get(
+      '--detect-open-handles'
+    ) as CommandLineFlagParameter;
+    const debugHeftReporter: CommandLineFlagParameter = taskSession.parametersByLongName.get(
+      '--debug-heft-reporter'
+    ) as CommandLineFlagParameter;
+    const disableCodeCoverage: CommandLineFlagParameter = taskSession.parametersByLongName.get(
+      '--disable-code-coverage'
+    ) as CommandLineFlagParameter;
+    const silent: CommandLineFlagParameter = taskSession.parametersByLongName.get(
+      '--silent'
+    ) as CommandLineFlagParameter;
+    const updateSnapshots: CommandLineFlagParameter = taskSession.parametersByLongName.get(
+      '--update-snapshots'
+    ) as CommandLineFlagParameter;
+
+    // Strings
+    const config: CommandLineStringParameter = taskSession.parametersByLongName.get(
+      '--config'
+    ) as CommandLineStringParameter;
+    const maxWorkers: CommandLineStringParameter = taskSession.parametersByLongName.get(
+      '--max-workers'
+    ) as CommandLineStringParameter;
+    const testTimeout: CommandLineStringParameter = taskSession.parametersByLongName.get(
+      '--test-timeout-ms'
+    ) as CommandLineStringParameter;
+    const findRelatedTests: CommandLineStringParameter = taskSession.parametersByLongName.get(
+      '--find-related-tests'
+    ) as CommandLineStringParameter;
+    const testNamePattern: CommandLineStringParameter = taskSession.parametersByLongName.get(
+      '--test-name-pattern'
+    ) as CommandLineStringParameter;
+    const testPathPattern: CommandLineStringParameter = taskSession.parametersByLongName.get(
+      '--test-path-pattern'
+    ) as CommandLineStringParameter;
+
+    taskSession.hooks.clean.tapPromise(PLUGIN_NAME, async (cleanOptions: IHeftTaskCleanHookOptions) => {
+      // Jest's cache is not reliable.  For example, if a Jest configuration change causes files to be
+      // transformed differently, the cache will continue to return the old results unless we manually
+      // clean it.  Thus we need to ensure that we always cleans the Jest cache.
+      cleanOptions.addDeleteOperations({ sourceFolder: taskSession.cacheFolder });
```

Comment:
Shouldn't Heft manage deletion of the cache folders?

Reply:
Only if you pass `--clean-cache`, which is not standard. Because we don't trust the Jest cache, we're manually deleting it ourselves, so I'm just keeping that behavior.